src/libcore/mem.rs

```diff
@@ -483,6 +483,16 @@ pub fn align_of_val<T: ?Sized>(val: &T) -> usize {
 /// ```
 #[inline]
 #[stable(feature = "needs_drop", since = "1.21.0")]
+#[rustc_const_unstable(feature = "const_needs_drop")]
+#[cfg(not(stage0))]
+pub const fn needs_drop<T>() -> bool {
+    intrinsics::needs_drop::<T>()
+}
+
+#[inline]
+#[stable(feature = "needs_drop", since = "1.21.0")]
+#[cfg(stage0)]
+/// Ceci n'est pas la documentation
```

Comment:
If #54601 merges first, you can drop this dummy non-const version.

Reply:
Thanks for the heads up. If #54601 merges first I'll be sure to sync and test with and without the dummy non-const version.